File: oms/src/main/webapp/resources/js/service/supplier_service.js
'use strict';

omsApp.factory('SupplierService', ['$http', '$q','CommonService', function($http, $q, CommonService){

	return {
		
			fetchAllSuppliers: function() {
					return $http.get(CommonService.getBaseUrl()+'/service/supplier/')
							.then(
									function(response){
										return response.data;
									}, 
									function(errResponse){
										console.error('Error while fetching suppliers');
										return $q.reject(errResponse);
									}
							);
			},
			
			addSupplier: function(supplier) {
				return $http.post(CommonService.getBaseUrl()+'/service/supplier/', supplier)
				.then(
						function(response){
							return response.data;
						}, 
						function(errResponse){
							console.error('Error while adding supplier');
							return $q.reject(errResponse);
						}
				);

			},
			updateSupplier: function(supplier) {
				return $http.put(CommonService.getBaseUrl()+'/service/supplier/', supplier)
				.then(
						function(response){
							return response.data;
						}, 
						function(errResponse){
							console.error('Error while updating supplier');
							return $q.reject(errResponse);
						}
				);

			},

			deleteSupplier: function(supplierId) {
				return $http.delete(CommonService.getBaseUrl()+'/service/supplier/'+supplierId)
				.then(
						function(response){
							return response.data;
						}, 
						function(errResponse){
							console.error('Error while fetching supplier count');
							return $q.reject(errResponse);
						}
				);

			},
			
			getSupplierCount: function() {
				return $http.get(CommonService.getBaseUrl()+'/service/supplier/count')
						.then(
								function(response){
									return response.data;
								}, 
								function(errResponse){
									console.error('Error while fetching supplier count');
									return $q.reject(errResponse);
								}
						);
			    
				}
		  
	};
	
	

}]);

File: oms/src/main/java/com/vjentrps/oms/service/SupplierService.java
package com.vjentrps.oms.service;

import java.util.List;

import com.vjentrps.oms.exception.OmsServiceException;
import com.vjentrps.oms.model.BasicInfo;
import com.vjentrps.oms.model.Supplier;

public interface SupplierService {
	
	 void addSupplier(Supplier supplier) throws OmsServiceException;
	
	 void deleteSupplier(long supplierId) throws OmsServiceException;
	
	 void updateSupplier(Supplier supplier) throws OmsServiceException;
	 
	 List<Supplier> listSuppliers() throws OmsServiceException;

	 Supplier getSupplierById(long supplierId) throws OmsServiceException;

	int getSupplierCount() throws OmsServiceException;

	List<BasicInfo> getSuppliersBasicInfo() throws OmsServiceException;

	Supplier getSupplierByName(String from) throws OmsServiceException;
	 
	
}
